### PR TITLE
Feature/add array builder

### PR DESCRIPTION
**Pull request by sebestyn168**
Hello,

This feature alllows to further script template in order to enable creating ArrayFields
The idea is to describe array in schema as follow
```
            SchemaBuilder.Create("TEST")
		.AddArray("Array TEST", f => f
			.Label("Label TEST")
			.Hints("Hints TEST")
			.Localizable()
			//.Disabled()
			//.Required()
			.AddString("String TEST", a => a
				.Required())
			.AddBoolean("Boolean TEST", a => a
				.Hints("Boolean hints TEST"))
			.AddDateTime("DateTime TEST", a => a
				.Hints("DateTime Hints TEST"))
			.AddNumber("Number TEST", a => a
				.Hints("Number Hints TEST"))
			.AddAssets("Logo TEST", a => a
				.MustBeImage()
				.Hints("Logo Hints TEST")))
			.Build();
```

### CODE REVIEW

File: backend/src/Squidex.Domain.Apps.Entities/Apps/Templates/Builders/ArrayFieldBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            get => base.field as UpsertSchemaField;
+            init => base.field = value;
+        }
+
+        public ArrayFieldBuilder(UpsertSchemaField field, CreateSchema schema)
+            : base(field, schema)
+        {
+        }
+
+        public new ArrayFieldBuilder Label(string? label) => base.Label(label) as ArrayFieldBuilder;
```
Comment (SebastianStehle) [inline backend/src/Squidex.Domain.Apps.Entities/Apps/Templates/Builders/ArrayFieldBuilder.cs:29]: Two things:

1. Can this not be solved with a generic base class: `ArrayFieldBuilder : FieldBuilder<ArrayFieldBuilder>`
2. Please do not use expression members, not a fan.

File: backend/src/Squidex.Domain.Apps.Entities/Apps/Templates/Builders/ArrayFieldBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+        public ArrayFieldBuilder AddBoolean(string name, Action<BooleanFieldBuilder> configure)
+        {
+            var field = AddField<BooleanFieldProperties>(name);
+            configure(new BooleanFieldBuilder(field, schema));
+            return this;
+        }
+
+        public ArrayFieldBuilder AddDateTime(string name, Action<DateTimeFieldBuilder> configure)
+        {
```
Comment (SebastianStehle) [inline backend/src/Squidex.Domain.Apps.Entities/Apps/Templates/Builders/ArrayFieldBuilder.cs:63]: Please make the formatting of this method consistent (similar to AddJson)

**Review by SebastianStehle — CHANGES_REQUESTED** [review]
Thank you very much. Just some minor change-requests.

**Comment by sebestyn168** [comment]
Hello,
I've added commits according to your requests.
I've ran some tests for the `ArrayFieldBuilder : FieldBuilder<T>` update and it works.

**Pull request merged**